### ISSUE TITLE
[FR] A markup language for appflowey

### ISSUE THREAD

**Issue by arromattic** [AppFlowy-IO/AppFlowy#7032]
### Description

Currently appflowey has two export option markdown and html . 
Markdown has no widely accepted standard  [example](https://gist.github.com/vimtaai/99f8c89e7d3d02a362117284684baa0f) . 
Appflowey also has features that like gallery that none of the flavor's of markdown cover . Exporting as regular markdown is destructive .  And for html it's not convenient either . 

### Suggestion 1 : Allow to export as Wikitext/mediawiki markup

https://en.wikipedia.org/wiki/Help:Wikitext 

Wikitext has way more formatting than markdown and can allow users to move or edit their note content in other apps and even in wikipedia (mediawiki)

### Suggestion 2 : A custom markup language for appflowey 

If it is not possible to implement export/import as wikitext a modified markdown can be created for appflowey .

### Impact

1.  Users can export data without losing formatting and blocks like Gallery , Image size , custom table width etc 
2. Easy to import/export data
3. Easy to convert into other file formats using pandoc

### Additional Context

Pandoc support if custom markup is implemented.

**Comment by LucasXu0** [comment]
@arromattic We have a JSON export mode in the experimental env, which include all document formats, such as image galleries, table widths, and more.

Here’s a sample. Does it meet your expectations?
[Simple_Table_V2.1.json](https://github.com/user-attachments/files/18224291/Simple_Table_V2.1.json)

**Comment by casainho** [comment]
Please no Markdown!! I came from Joplin that is good app, although Markdown makes is a junk, like tables are really basic and I think is due to limitations on Markdown.

**Comment by arromattic** [comment]
The json format is a significant improvement over plain markdown .  But is rather difficult for average user to read/edit manually like wikitext . Are all the syntaxes will be specified ? like color , mathblock , italic etc . If yes i think that's most likely enough for now as users don't have to worry about missing formattings when exporting/importing . 

 I do think a new file format should be created and the syntaxes should be specified instead of generic json so users can easily identify it and easily convert into other format using a script or pandoc (assuming someone adds it).
 
 
1. Is it possible to import the json to appflowy ? 
2. Does the json export only covers table like how notion has separate file for table and markdown for notes or includes everything in a page ?

**Comment by arromattic** [comment]
> Please no Markdown!! I came from Joplin that is good app, although Markdown makes is a junk, like tables are really basic and I think is due to limitations on Markdown.

I am also a user of joplin and joplins markdown export is what made me create this issue . 

In joplin there are a few markdown export problems : 
1. You can adjust image size on the note but on export markdown doesn't support image size 
2. You can download various plug-ins to extend the markdown but once you export , you have no idea of what you used to use .

**Comment by casainho** [comment]
@arromattic , I agree that is very important to be able to easily export and import the full data for backup reasons and in a way that is easy to process.
On Joplin, I usually export to their JEX file and import if needed (usually when updating to new server or PC app versions).

What I mainly miss from Joplin, is to have the full files locally, so if I am offline I always have the data AS also, is very fast to open the files and even edit the files and knowing they will be updated on the note!!

**Comment by LucasXu0** [comment]
> Is it possible to import the json to appflowy ?

If we support exporting JSON, then the import functionality should also be supported.

> Does the json export only covers table like how notion has separate file for table and markdown for notes or includes everything in a page ?

The JSON export will include everything on a page. By 'table,' do you mean the table block or the database block?

**Comment by arromattic** [comment]
> The JSON export will include everything on a page. By 'table,' do you mean the table block or the database block?

The database block .

**Comment by rsenna** [comment]
Hey! I’m completely new to AppFlowy — just found out about it through a news article — but I’m already super impressed by the UX and codebase. Even though I’m new to this app, I’m a seasoned dev and a “hardcore” Obsidian user, so I thought I’d share my 2 cents on the proposals here:
- Wikitext: Big thumbs up from me! 😅
- Markdown: While not “standardised” (plenty of flavours out there), it’s the standard in the [Second Brain](https://fortelabs.com/blog/basboverview/) and [Personal Knowledge Management](https://www.wikiwand.com/en/articles/Personal_knowledge_management) world. So honestly, I think strong, advanced Markdown options for import/export are a **must**.

For example, you could use [Frontmatter](https://www.markdowntoolbox.com/blog/common-front-matter/) to add custom metadata to a document. With that metadata, you could define styling, block positions, or other properties. Even if how AppFlowy uses the metadata isn’t “standard,” the document would still be valid Markdown, making it easy to interface with other tools.

Going down a “custom file format” route feels like it would clash with the Open-Source (and Open Standards) ethos this project seems to embrace.

Just my two cents! Keep up the awesome work! 👏

**Comment by Matt2012** [comment]
This is the AI age not giving frontmatter markdown support first class support would be crazy for use cases you can only start to imagine.

That doesn't mean limiting the interface to markdown. Perfectly acceptable to lose some fidelity as long as page meta data and    readable text is contained in the markdown.

**Comment by beedell-roke** [comment]
Almost all of the content of this thread is incorrect. Markdown is standardised in CommonMark. However, it lacks many features that people want. Luckily, those features are provided by CM's WHATWG HTML5 LS integration. However, because indentation across lines is not supported by CM, I propose that AppFlowy's markup language be HTML. For anyone who cannot write it, an LLM can generate it, and for anyone who can write it, nothing comprehensively more versatile exists.
